akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java

```diff
@@ -0,0 +1,131 @@
+import java.util.regex.Pattern;
+
+import org.junit.Test;
+
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.HttpResponse;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.headers.Host;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.server.RequestContext;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import java.util.function.Function;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import akka.http.javadsl.model.Uri;
+import akka.http.javadsl.model.headers.Location;
+import akka.http.javadsl.server.directives.DebuggingDirectives;
+import akka.http.javadsl.server.directives.RouteDirectives;
+import akka.event.Logging;
+import akka.event.Logging.LogLevel;
+import akka.http.javadsl.server.directives.LogEntry;
+import java.util.List;
+import akka.http.scaladsl.server.Rejection;
+
+import static akka.event.Logging.InfoLevel;
+import java.util.stream.Collectors;
+import java.util.Optional;
+
+public class DebuggingDirectivesExamplesTest extends JUnitRouteTest {
+
+  @Test
+  public void testLogRequest() {
+    //#logRequest
+    // logs request with "get-user"
+    final Route routeBasicLogRequest = get(() -> 
+      logRequest("get-user", () -> complete("logged")));
+    
+    // logs request with "get-user" as Info
+    final Route routeBasicLogRequestAsInfo = get(() -> 
+      logRequest("get-user", InfoLevel(), () -> complete("logged")));
+
+    // logs just the request method at info level
+    Function<HttpRequest, LogEntry> requestMethodAsInfo = (request) -> 
+      LogEntry.create(request.method().toString(), InfoLevel());
+
+    final Route routeUsingFunction = get(() -> 
+      logRequest(requestMethodAsInfo, () -> complete("logged")));
+
+    // tests:
+    testRoute(routeBasicLogRequest).run(HttpRequest.GET("/"))
+      .assertEntity("logged");
+    //#logRequest
+  }
+
+  @Test
+  public void testLogRequestResult() {
+    //#logRequestResult
+    // using logRequestResult
+
+    // handle request to optionally generate a log entry
+    BiFunction<HttpRequest, HttpResponse, Optional<LogEntry>> requestMethodAsInfo = 
+      (request, response) ->
+        (response.status().isSuccess())  ? 
+            Optional.of(
+              LogEntry.create(
+                request.method().toString() + ":" + response.status().intValue(), 
+                InfoLevel()))
+          : Optional.empty(); // not a successful response
+
+    // handle rejections to optionally generate a log entry
+    BiFunction<HttpRequest, List<Rejection>, Optional<LogEntry>> rejectionsAsInfo = 
+      (request, rejections) ->
+        (!rejections.isEmpty())  ? 
+          Optional.of(
+            LogEntry.create(
+                rejections
+                .stream()
+                .map(Rejection::toString)
+                .collect(Collectors.joining(", ")), 
+              InfoLevel()))
+          : Optional.empty(); // no rejections
```

Comment:
wouldn't these be easier to read using Optional's orElseGet?  https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseGet-java.util.function.Supplier-